aiosmtpd/smtp.py

```diff
@@ -751,8 +758,9 @@ async def smtp_MAIL(self, arg: str) -> None:
             return
         address, params = self._getaddr(arg)
         if address is None:
-            await self.push(syntaxerr)
-            return
+            return await self.push("553 5.1.3 Error: malformed address")
+        if not address:
+            return await self.push(syntaxerr)
```

Comment:
Need to differ between "cannot parse email" (`address is None`) and "email is parsed but is empty" (`address == ""`)

Also I just realized I can do `return await` to make the code vertically more compact, as both `self.push` and this method return None 😀